registryprotocols/ociclient/client.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+
+package ociclient
+
+import (
+	"context"
+)
+
+// OCIClient implements a high level interface giving you access to a Docker / OCI Registry.
+type OCIClient interface {
```

Comment:
General comment re: names in this package, you can drop the redundant `OCI` portion. Code that imports this package will reference this as `ociclient.OCIClient`, which could be simply `ociclient.Client`.

https://go.dev/wiki/CodeReviewComments#package-names